core/src/impl/Kokkos_GraphNodeThenImpl.hpp

```diff
@@ -21,35 +21,72 @@
 #include <impl/Kokkos_GraphImpl_fwd.hpp>
 
 namespace Kokkos::Impl {
+// Concept for a tag class.
+template <typename T>
+struct is_tag_class
+    : std::integral_constant<
+          bool, std::conjunction_v<std::is_class<T>, std::is_empty<T>,
+                                   std::is_trivially_default_constructible<T>,
+                                   std::is_trivially_copy_constructible<T>,
+                                   std::is_trivially_move_constructible<T>,
+                                   std::is_trivially_destructible<T>>> {};
 
+template <typename T>
+inline constexpr bool is_tag_class_v = is_tag_class<T>::value;
+}  // namespace Kokkos::Impl
+
+namespace Kokkos::Experimental {
+template <typename WorkTag>
+struct ThenPolicy {
+  static_assert(Kokkos::Impl::is_tag_class_v<WorkTag> ||
+                std::is_void_v<WorkTag>);
+
+  using work_tag = WorkTag;
+
+  std::string label;
```

Comment:
Why do we have a label member in the policy?
This would be another departure from other existing execution policies.

Reply:
Oups, indeed ! Let me correct this :wink: